src/platform/client/client_platform_linux.cpp

```diff
@@ -24,8 +24,9 @@ namespace mcp = multipass::cli::platform;
 
 void mcp::open_multipass_shell(const QString& instance_name)
 {
-    QProcess::startDetached(
-        "xterm", {"-title", instance_name, "-j", "-e", QString("multipass shell %1 || read").arg(instance_name)});
+    if (!instance_name.isEmpty())
```

Comment:
Could this check be moved to the call site, please? 

I believe it's the job of the GUI and not so much this util function to avoid opening a shell when primary is disabled (this function is better regarded as having a(n undocumented) pre-condition that the instance name is valid). 

This can have an assert instead, if you like.